src/clib/pio_rearrange.c

```diff
@@ -738,6 +746,7 @@ int rearrange_comp2io(const iosystem_desc_t ios, io_desc_t *iodesc, void *sbuf,
         sendtypes[i] =  MPI_DATATYPE_NULL;
     }
 
+    /* ??? */
```

Comment:
If this io proc will exchange data with compute tasks create a MPI DataType for that exchange.